src/explosion.cpp

```diff
@@ -356,10 +357,9 @@ static std::vector<tripoint> shrapnel( const tripoint &src, int power,
     // TODO: Calculate range based on max effective range for projectiles.
     // Basically bisect between 0 and map diameter using shrapnel_calc().
     // Need to update shadowcasting to support limiting range without adjusting initial distance.
-    const tripoint start = { 0, 0, src.z };
-    const tripoint end = { g->m.getmapsize() *SEEX, g->m.getmapsize() *SEEY, src.z };
+    const tripoint_range area = g->m.points_on_zlevel( src.z );
 
-    g->m.build_obstacle_cache( start, end, obstacle_cache );
+    g->m.build_obstacle_cache( area.min(), area.max() + tripoint( 1, 1, 0 ), obstacle_cache );
```

Comment:
```suggestion
    g->m.build_obstacle_cache( area.min(), area.max() + tripoint_south_east, obstacle_cache );
```
I'm not sure how I feel about this, we might want to revisit how we handle this warning, but for now this should silence the warning.